pkg/network/server.go

```diff
@@ -387,10 +387,28 @@ func (s *Server) ConnectedPeers() []string {
 // while itself dealing with peers management (handling connects/disconnects).
 func (s *Server) run() {
 	go s.runProto()
-	for {
-		if s.PeerCount() < s.MinPeers {
+	for loopCnt := 0; ; loopCnt++ {
+		var (
+			netSize = s.discovery.NetworkSize()
+			// "Optimal" number of peers.
+			optimalN = s.discovery.GetFanOut() * 2
```

Comment:
Why `optimalN` is Fout*2?

Reply:
Multiplying by 2 can't be bad! It's all heuristics, there is no way around it. But it seems to be legit to me, we have ~100-200 nodes on the mainnet, so the fan out magic number is around 12-13 and we want ~20-30 connections to have good connectivity.